src/story.ts

```diff
@@ -259,6 +259,13 @@ export type LoaderFunction<TRenderer extends Renderer = Renderer, TArgs = Args>
   context: StoryContextForLoaders<TRenderer, TArgs>
 ) => Promise<Record<string, any> | void> | Record<string, any> | void;
 
+type Awaitable<T> = T | PromiseLike<T>;
+export type CleanupCallback = () => Awaitable<unknown>;
+
+export type BeforeEach<TRenderer extends Renderer = Renderer, TArgs = Args> = (
+  context: StoryContextForLoaders<TRenderer, TArgs>
```

Comment:
is StoryContextForLoaders the right type? or should it be StoryContext?

Reply:
changed indeed